wsagent/che-core-api-project/src/test/java/org/eclipse/che/api/project/server/ProjectServiceTest.java

```diff
@@ -928,6 +939,50 @@ public void testDeleteProject() throws Exception {
     }
 
     @Test
+    public void testDeleteProjectsConcurrently() throws Exception {
+        int threadNumber = 100;
+        ExecutorService executor = Executors.newFixedThreadPool(threadNumber);
+        CountDownLatch countDownLatch = new CountDownLatch(threadNumber);
+        List<Future<ContainerResponse>> futures = new LinkedList<>();
```

Comment:
Okay, obviously for me it is not a big deal to replace a `LinkedList` with an `ArrayList`. But if you indeed distinguish velocity of ArrayList and LinkedList and optimization critically matters here then tell me why do you insist on using an `ArrayList` while java array will be even more efficient?

I guess if I know the size of an array I definitely should use simple java array (taking into account that I need only set element and get iterator).

I need to know how do you see the difference between well-optimized and over-optimized code, how do you see where optimization matters more than convenience. Where can I find any rules or conventions or you just impose your preferences? 😆